Content.Server/Forensics/Systems/ForensicsSystem.cs

```diff
@@ -101,17 +104,51 @@ public void CopyForensicsFrom(ForensicsComponent src, EntityUid target)
             }
         }
 
-        private void OnAfterInteract(EntityUid uid, CleansForensicsComponent component, AfterInteractEvent args)
+        // Try: "Entity<CleansForensicsComponent> cleanitemcomp" or something!
+        private void OnAfterInteract(Entity<CleansForensicsComponent> cleanForensicsEntity, ref AfterInteractEvent args)
         {
             if (args.Handled || !args.CanReach)
                 return;
 
-            if (!TryComp<ForensicsComponent>(args.Target, out var forensicsComp))
+            args.Handled = StartClean(cleanForensicsEntity, args.User, args.Target);
+        }
+
+        private void OnUtilityVerb(Entity<CleansForensicsComponent> entity, ref GetVerbsEvent<UtilityVerb> args)
+        {
+            if (!args.CanInteract || !args.CanAccess || !args.CanInteract)
                 return;
 
-            if((forensicsComp.DNAs.Count > 0 && forensicsComp.CanDnaBeCleaned) || (forensicsComp.Fingerprints.Count + forensicsComp.Fibers.Count > 0))
+            // These need to be set outside for the anonymous method!
+            var user = args.User;
+            var target = args.Target;
+
+            var verb = new UtilityVerb()
             {
-                var doAfterArgs = new DoAfterArgs(EntityManager, args.User, component.CleanDelay, new CleanForensicsDoAfterEvent(), uid, target: args.Target, used: args.Used)
+                Act = () => StartClean(entity, user, target),
+                IconEntity = GetNetEntity(entity),
+                Text = Loc.GetString(Loc.GetString("forensics-verb-text")),
+                Message = Loc.GetString(Loc.GetString("forensics-verb-message"))
+            };
+
+            args.Verbs.Add(verb);
+        }
+
+        /// <summary>
+        ///     Attempts to clean the given item with the given CleansForensics entity.
+        /// </summary>
+        /// <param name="cleanForensicsEntity">The entity that is being used to clean the target.</param>
+        /// <param name="user">The user that is using the cleanForensicsEntity.</param>
+        /// <param name="target">The target of the forensics clean.</param>
+        /// <returns>True if the target can be cleaned and has some sort of DNA or fingerprints / fibers and false otherwise.</returns>
+        public bool StartClean(Entity<CleansForensicsComponent> cleanForensicsEntity, EntityUid user, EntityUid? target)
```

Comment:
Every failure state (return false) should have user feedback in the form of a red popup (method has a popup type param), i think that the verb shouldnt even be added if the target doesnt have forensics component but that could make it less discoverable so i leave that call to you

Reply:
I think the system is already kind of hidden enough so I personally would go for leaving the verb for all targets! I didn't even think of having the popup failure though so I added that.

Let me know if that looks good or if I should just remove the verb from non cleanable stuff!